modules/apps/transfer/client/cli/tx.go

```diff
@@ -85,62 +80,18 @@ corresponding to the counterparty channel. Any timeout set to 0 is disabled.`),
 				return err
 			}
 
-			// if the timeouts are not absolute, retrieve latest block height and block timestamp
-			// for the consensus state connected to the destination port/channel.
-			// localhost clients must rely solely on local clock time in order to use relative timestamps.
+			// NOTE: relative timeouts using block height are not supported.
+			// if the timeouts are not absolute, CLI users rely solely on local clock time in order to use relative timestamps.
 			if !absoluteTimeouts {
-				clientRes, err := channelutils.QueryChannelClientState(clientCtx, srcPort, srcChannel, false)
-				if err != nil {
-					return err
-				}
-
-				var clientState exported.ClientState
-				if err := clientCtx.InterfaceRegistry.UnpackAny(clientRes.IdentifiedClientState.ClientState, &clientState); err != nil {
-					return err
-				}
-
-				clientHeight, ok := clientState.GetLatestHeight().(clienttypes.Height)
-				if !ok {
-					return fmt.Errorf("invalid height type. expected type: %T, got: %T", clienttypes.Height{}, clientState.GetLatestHeight())
-				}
-
-				var consensusState exported.ConsensusState
-				if clientState.ClientType() != exported.Localhost {
-					consensusStateRes, err := clientutils.QueryConsensusState(clientCtx, clientRes.IdentifiedClientState.ClientId, clientHeight, false, true)
-					if err != nil {
-						return err
-					}
-
-					if err := clientCtx.InterfaceRegistry.UnpackAny(consensusStateRes.ConsensusState, &consensusState); err != nil {
-						return err
-					}
-				}
-
 				if !timeoutHeight.IsZero() {
-					absoluteHeight := clientHeight
-					absoluteHeight.RevisionNumber += timeoutHeight.RevisionNumber
-					absoluteHeight.RevisionHeight += timeoutHeight.RevisionHeight
-					timeoutHeight = absoluteHeight
+					return errors.New("relative timeouts using block height is not supported")
 				}
 
-				// use local clock time as reference time if it is later than the
-				// consensus state timestamp of the counterparty chain, otherwise
-				// still use consensus state timestamp as reference.
-				// for localhost clients local clock time is always used.
+				// use local clock time as reference time for calculating timeout timestamp.
 				if timeoutTimestamp != 0 {
```

Comment:
I think the code should actually be:
```go
if !absolute {
    if !timeoutHeight.IsZero() {
        return err
    }
    
    if timeoutTimestamp == 0 {
        return err
    }
    
    now := time.Now().UnixNano()
    if now <= 0 {
        return errors.New("local clock time is not greater than Jan 1st, 1970 12:00 AM")
    }
    
    timeoutTimestamp = uint64(now) + timeoutTimestamp
}
```

Reply:
Happy to go with this, but if timeouts are not absolute - someone would also have to explicitly provide a zero value timestamp to trigger the err here. Because will be default 10 mins if not overridden. Still tho, returning the err in this case makes perfect sense. Will implement as suggested, ty!

Reply:
Plus one level of if indentation is removed here 😂